File: backend/placement-management/src/main/java/com/placement/management/service/AuthService.java
package com.placement.management.service;

import com.placement.management.dto.*;

/**
 * Authentication service contract covering registration, login, token refresh,
 * forgot/reset password, change password, and logout.
 *
 * @author feature/auth
 */
public interface AuthService {

    /**
     * Registers a new user and returns JWT tokens.
     *
     * @param request registration payload
     * @return {@link AuthResponse} with access token, refresh token, and user info
     * @throws IllegalArgumentException if the email is already taken or ADMIN role is requested
     */
    AuthResponse register(RegisterRequest request);

    /**
     * Authenticates a user and returns JWT tokens.
     *
     * @param request login payload
     * @return {@link AuthResponse} with access token, refresh token, and user info
     */
    AuthResponse login(LoginRequest request);

    /**
     * Issues a new access token from a valid refresh token.
     *
     * @param request refresh-token payload
     * @return new {@link AuthResponse}
     */
    AuthResponse refreshToken(RefreshTokenRequest request);

    /**
     * Initiates the forgot-password flow: generates a reset token, stores it, and logs
     * the reset URL.
     * the reset URL (email delivery is handled by {@link EmailService}).
     *
     * @param request contains the user's email
     */
    void forgotPassword(ForgotPasswordRequest request);

    /**
     * Validates the password-reset token and updates the user's password.
     *
     * @param request contains the token and new password
     * @param request contains the token, new password, and confirmation
     */
    void resetPassword(ResetPasswordRequest request);

    /**
     * Changes the password for the currently authenticated user.
     *
     * @param email   the authenticated user's email
     * @param request contains current password and new password
     * @param email             the authenticated user's email
     * @param request           contains current password, new password, and confirmation
     */
    void changePassword(String email, ChangePasswordRequest request);

    /**
     * Logs out the user.
     * Logs out the user (client-side token invalidation; extend to a deny-list if needed).
     *
     * @param token the JWT access token to invalidate
     */
    void logout(String token);
}

File: frontend/placement-frontend/src/services/authService.js
import api from './api';

/**
 * Authentication service — wraps all /auth API endpoints.
 *
 * All methods return the Axios response directly; callers should
 * access `.data.data` for the payload and `.data.message` for messages.
 */
const authService = {

  /**
   * Register a new user (STUDENT or PLACEMENT_OFFICER).
   */
  register: (name, email, password, role) =>
    api.post('/auth/register', { name, email, password, role }),

  /**
   * Login with email and password. Returns access + refresh tokens + user info.
   */
  login: (email, password) =>
    api.post('/auth/login', { email, password }),

  /**
   * Logout — instructs server to log the event; client clears tokens.
   */
  logout: () =>
    api.post('/auth/logout'),

  /**
   * Refresh access token using a stored refresh token.
   */
  refreshToken: (refreshToken) =>
    api.post('/auth/refresh', { refreshToken }),

  /**
   * Initiate forgot-password flow. Server sends reset link.
   */
  forgotPassword: (email) =>
    api.post('/auth/forgot-password', { email }),

  /**
   * Complete password reset with the token from the email link.
   */
  resetPassword: (token, newPassword, confirmPassword) =>
    api.post('/auth/reset-password', { token, newPassword, confirmPassword }),

  /**
   * Get the currently authenticated user's profile.
   */
  getCurrentUser: () =>
    api.get('/auth/me'),

  /**
   * Change password for the logged-in user.
   */
  changePassword: (currentPassword, newPassword, confirmPassword) =>
    api.post('/auth/change-password', { currentPassword, newPassword, confirmPassword }),
};

export default authService;
